ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,128 @@ actor_system_config:
+
+### Per-component memory limits
+
+There are two different types of components inside {{ ydb-short-name }}.
+
+The first type components, or cache components, acts like caches, for example storing the last recently used data. Each of cache components has minimum and maximum memory limit thresholds, allowing them to change their capacity dynamically based on current {{ ydb-short-name }} process consumption.
+
+The second type components, or activity components, allocate memory for specific activities, such as query execution or the [compaction](../../concepts/glossary.md#compaction) process. Each of activity components has a fixed memory limit. There is also an additional total memory limit for these activities, from which they attempt to consume the needed memory.
+
+Many other auxiliary components and processes operate alongside the {{ ydb-short-name }} process, consuming memory. Currently, these components do not have any memory limits.
+
+#### Cache components memory limits
+
+Cache components are:
+
+- Shared cache
+- MemTable
+
+Each of cache components' limits is dynamically recalculated every second so that each component consumes memory proportionally to its limit thresholds, and the total consumed memory stays around the target memory utilization.
+
+Cache components' minimum memory limit threshold isn't reserved, meaning that the memory remains available until it is actually used. However, once this memory is filled, the components typically retain the data, operating within their current memory limit. Consequently, the sum of cache components minimum memory limits is expected to be less than the target memory utilization.
+
+If needed, both the minimum and maximum thresholds should be overridden; otherwise, a missing threshold will have a default value.
+
+Example of the `memory_controller_config` section with specified shared cache limits:
+
+```yaml
+memory_controller_config:
+  shared_cache_min_percent: 10
+  shared_cache_max_percent: 30
+```
+
+#### Activity components memory limits
+
+Activity components are:
+
+- KQP
+
+The memory limit for each of the activity components specifies the maximum amount of memory it can attempt to use. However, to prevent {{ ydb-short-name }} process from exceeding the soft memory limit, the total consumption of activity components is further limited by an additional limit, named as the activities memory limit. If the total memory usage of the activity components exceeds this limit, any additional memory requests will be denied.
+
+As a result, while the combined individual limits of the activity components might collectively exceed the activities memory limit, each component's individual limit should be less than this overall cap. Additionally, the sum of the minimum memory limits for the cache components plus the activities memory limit needs to be less than the soft memory limit.
+
+There are some other activity components, currently they do not have any individual memory limits.
```

Comment:
```suggestion
#### Activity components memory limits

The activity components include:

- KQP

The memory limit for each activity component specifies the maximum amount of memory it can attempt to use. However, to prevent the {{ ydb-short-name }} process from exceeding the soft memory limit, the total consumption of activity components is further constrained by an additional limit known as the activities memory limit. If the total memory usage of the activity components exceeds this limit, any additional memory requests will be denied.

As a result, while the combined individual limits of the activity components might collectively exceed the activities memory limit, each component's individual limit should be less than this overall cap. Additionally, the sum of the minimum memory limits for the cache components, plus the activities memory limit, must be less than the soft memory limit.

There are some other activity components that currently do not have individual memory limits.
```